pretixdroid/app/src/main/java/eu/pretix/pretixdroid/ui/EventinfoActivity.java

```diff
@@ -0,0 +1,338 @@
+
+                JSONArray items = result.getJSONArray("items");
+                for (int i = 0; i < items.length(); i++) {
+                    EventItemCardItem eici = new EventItemCardItem(items.getJSONObject(i));
+                    eia.addItem(eici);
+                }
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * implementation of an adapter for a listview to hold EventCards and EventItemCards
+     */
+    public class EventItemAdapter extends BaseAdapter {
+        public static final int TYPE_EVENTCARD = 0;
+        public static final int TYPE_EVENTITEMCARD = 1;
+        private static final int MAX_TYPES = 2;
+
+        private final ArrayList<EventinfoListItem> mData = new ArrayList<>();
+        public final LayoutInflater mInflater;
+
+        public EventItemAdapter(Context ctx) {
+            mInflater = LayoutInflater.from(ctx);
+        }
+
+        public void addItem(EventinfoListItem item) {
+            this.mData.add(item);
+            notifyDataSetChanged();
+        }
+
+        @Override
+        public int getItemViewType(int position) {
+            return this.mData.get(position).getType();
+        }
+
+        @Override
+        public int getViewTypeCount() {
+            return MAX_TYPES;
+        }
+
+        @Override
+        public int getCount() {
+            return mData.size();
+        }
+
+        @Override
+        public EventinfoListItem getItem(int position) {
+            return mData.get(position);
+        }
+
+        @Override
+        public long getItemId(int position) {
+            return position;
+        }
+
+        @Override
+        public View getView(int position, View convertView, final ViewGroup parent) {
+            if (convertView == null) {
+                EventinfoListItem item = this.getItem(position);
+                return item.getCard(mInflater, parent);
+            } else {
+                ((EventinfoListItem) convertView.getTag()).fillView(convertView, mInflater, parent);
+                return convertView;
+            }
+
+        }
+
+    }
+
+    /**
+     * interface of all CardItems that are displayed on the eventinfo page
+     */
+    public interface EventinfoListItem {
+        /**
+         * returns an integer for the adapter to distinguish between cards
+         * @return an integer for the adapter to distinguish between cards
+         */
+        int getType();
+
+        /**
+         * returns a newly inflated card with the content of this item
+         * @param inflater the inflater to use
+         * @param parent the parent ViewGroup
+         * @return a newly inflated card with the content of this item
+         */
+        View getCard(LayoutInflater inflater, ViewGroup parent);
+
+        /**
+         * returns a recycled view filled with the contents of this item
+         * @param view a recycled view filled with the contents of this item
```

Comment:
I don't think its necessary to write all of this twice. This makes it error-prone to change the docs later, which is worse than to omit one of the two (I don't really care which one)